src/audio/data_blob.c

```diff
@@ -272,6 +272,138 @@ int comp_data_blob_set(struct comp_data_blob_handler *blob_handler,
 	return 0;
 }
 
+int ipc4_comp_data_blob_set(struct comp_data_blob_handler *blob_handler,
+			    bool first_block,
+			    bool last_block,
+			    uint32_t data_offset,
+			    char *data)
+{
+	int ret;
+	int valid_data_size;
+
+	if (!blob_handler) {
+		comp_err(blob_handler->dev,
+			 "ipc4_comp_data_blob_set(): blob_handler is NULL!");
+		return -EINVAL;
+	}
+
+	comp_dbg(blob_handler->dev,
+		 "ipc4_comp_data_blob_set(): data_offset = %d",
+		 data_offset);
+
+	/* In single blob mode the component can not be reconfigured if
+	 * the component is active.
+	 */
+	if (blob_handler->single_blob &&
+	    blob_handler->dev->state == COMP_STATE_ACTIVE) {
+		comp_err(blob_handler->dev,
+			 "ipc4_comp_data_blob_set(): On the fly updates forbidden!");
+		return -EBUSY;
+	}
```

Comment:
How would this work for EQs ? I think the FIR/IIR can support this today with IPC3. @singalsu can you comment ?

Reply:
@iganakov ping ?

Reply:
@iganakov @jsarha can you confirm if the above change blocks stream runtime "on the fly" updates. It looks like from the error message that runtime updates are not supported for IPC4 ?

Reply:
@lgirdwood IPC4 supports 'on the fly' updates. But in case of very large configs it depends on module implementation. `ipc4_comp_data_blob_set()` is used mostly to handle such large configs so I would rather leave this error.

Reply:
but we should not put the error check in the core if it's a per module implementation feature. i.e the module should return an error (if it does not support on the fly) and core just passes this error up the stack to host SW.

Reply:
> How would this work for EQs ? I think the FIR/IIR can support this today with IPC3. @singalsu can you comment ?

Yep, most of the components do support on the fly update. It's important to help the audio tuning process plus if some endpoint  has multiple operation modes.

Please add the on the fly reject to be set from keyword component.

Reply:
@singalsu @lgirdwood  Keyword component supports "on the fly" updates and its config is relatively small (0x1000) so it doesn't need such reject to be implemented. I'm just removing this error check from `ipc4_comp_data_blob_set()`